stetho/src/main/java/com/facebook/stetho/inspector/elements/ChainedDescriptor.java

```diff
@@ -53,5 +53,5 @@
  * {@link AbstractChainedDescriptor}) which solves all of these issues for you.<p/>
  */
 public interface ChainedDescriptor {
-  void setSuper(Descriptor superDescriptor);
+  void setSuper(Descriptor<Object> superDescriptor);
```

Comment:
actually ... it'd be great if we could somehow make it so that this is some form like `Descriptor<? super E>`, so that `AbstractChainedDescriptor<E>` can enforce type safety better

It may be best to leave `ChainedDescriptor#setSuper` as accepting a raw `Descriptor` and do the `<? super E>` thing only in the derived class. (java generics are weird ... that might even work)